### PR TITLE
[DBEX-64149] Update accordion and edit components on forms review page

### PR DESCRIPTION
**Pull request by zack**
## Summary

- The review page was using custom components for the "accordion"-style effect. These components need to be migrated to the design system. The specific impetus is a contrast issue.
- The accordion item buttons were not meeting AA accessibility guidelines when in `:active` state.
- The solution is to migrate the components to use `<va-accordion>`.
- This work is being done under the color of the Non-Disability Benefits Experience team. This team owns some forms being impacted by this work, but does not own the system or specific components being modified in this PR.
- This work is not behind Flipper.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/64149

## Testing done

- Unit tests were updated to work with the new changes. Some tests are no longer needed and were removed (these decisions are explained in comments on the PR).
- Manual testing shows the features working properly with the new components (see screenshots).
- Testing can be done by navigating to the end of any form.

### Accordion Active State

<details>
<summary>
Before
</summary>

image
</details>

<details>
<summary>
After
</summary>

image
</details>

### Accordion Error State
<details>
<summary>
Before
</summary>

deps-contrast-forms-error-old
</details>

<details>
<summary>
After
</summary>

deps-contrast-forms-error-fixed
</details>

### Accordion scroll and focus behavior
- All autoscrolling has been removed from the page
- Autofocus behavior has been maintained
<details>
<summary>
Video
</summary>

https://github.com/department-of-veterans-affairs/vets-website/assets/2482403/920473a7-d114-4cbc-9f7c-f229885df859
</summary>
</details>

### Button Active State
<details>
<summary>
Before
</summary>

image
</details>

<details>
<summary>
After
</summary>

image

</details>

### Video
<details>
<summary>
Final product
</summary>

https://github.com/department-of-veterans-affairs/vets-website/assets/2482403/7f10894a-2332-4bf5-8e14-1b21ee640646
</details>

## What areas of the site does it impact?

**This will impact the "Review" page on every form using Platform's Form System.** 

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/edu-benefits/1990s/analytics/analytics-events.js
Comment (zack) [inline src/applications/edu-benefits/1990s/analytics/analytics-events.js:21]: I checked with the GA team and these actions are not needed.

File: src/applications/toe/tests/fixtures/data/form-submission-test-data.js
Comment (zack) [inline src/applications/toe/tests/fixtures/data/form-submission-test-data.js:69]: You will see lots of code in this PR removed that has to do with open chapters. Redux code, fixture code, actions, etc. All of this code supported keeping track of and firing events on the review page chapter accordions. Since we're migrating to the `<va-accordion>` which handles its own state and events, we can take it all out.

File: src/platform/forms-system/src/js/review/ObjectField.jsx
Comment (zack) [inline src/platform/forms-system/src/js/review/ObjectField.jsx:165]: This is for the part of the ticket about the edit button accessibility. This and the line on 190 have to do with that. The rest of the PR is for the accordions.

File: src/platform/forms-system/src/js/review/ReviewChapters.jsx
Comment (zack) [inline src/platform/forms-system/src/js/review/ReviewChapters.jsx:50]: Moved all of this into the chapter component. It didn't need to be in the parent.

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (zack) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:37]: This component needs to keep track of its own open state now that it is handling the scroll behavior.

File: src/platform/forms-system/test/js/review/ReviewChapters.unit.spec.jsx
Comment (zack) [inline src/platform/forms-system/test/js/review/ReviewChapters.unit.spec.jsx:57]: This logic is now handled by `<va-accordion>`.

File: src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx
Comment (zack) [inline src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx:718]: This is replaced by a `<va-accordion-item>` which I don't believe needs to be tested, as it is library code.

**Review by zack — COMMENTED** [review]
Explanation for reviewers.

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (tblackwe) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:348]: I'd like to avoid using `data-testid` unless it is required. Are we able to test this without?
My reasoning comes from the testing-library guiding principles, we should aim to test how users interact with the code. `getByTestId` uses the `data-testid` hook to locate the exact spot in the page, which is not how a user would interact with the same div. 

For reference: React Testing Query Priority (https://testing-library.com/docs/queries/about#priority)

Comment (zack) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:321]: :) I generally agree! But here we don't really have anything useful to query for, and since this is only used when we're testing that we're _not_ rendering anything inside this container element I felt it was a fair concession to make. But I'm always happy to consider a different viewpoint if you have something to propose!

File: config/webpack.config.js
Comment (Midge-dev) [inline config/webpack.config.js:1]: We don't want to commit this, it'll break build

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (Midge-dev) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:110]: Are you checking for null/undefined here? This is probably a good area for nullish coalescing `??` @tblackwe reminded me of :)

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (Midge-dev) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:328]: Curious what the "t" is for

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (Midge-dev) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:344]: :rocket:

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (Midge-dev) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:399]: Reuse `subHeader` here :)

File: src/platform/forms-system/src/js/utilities/ui/index.js
Comment (Midge-dev) [inline src/platform/forms-system/src/js/utilities/ui/index.js:89]: Maybe use optional chaining here in case `shadowRoot` doesn't exist

`focusElement(focusTarget?.shadowRoot?.querySelector(shadowTarget));`

File: src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx
Comment (Midge-dev) [inline src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx:927]: :fire:

**Review by Midge-dev — CHANGES_REQUESTED** [review]
Minor things then I can approve :) Nice work!

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:110]: Logical or (`||`)takes care of that:

```
Welcome to Node.js v14.15.0.
Type ".help" for more information.
> undefined || '';
''
> null || '';
''
```

This isn't new logic, I'm just moving it to a reusable place. :)

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:328]: Not sure, but didn't want touch it. :)

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:399]: ooh yes thank you

File: src/platform/forms-system/src/js/utilities/ui/index.js
Comment (zack) [reply src/platform/forms-system/src/js/utilities/ui/index.js:89]: Sure, good check

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (tblackwe) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:321]: Thanks! I'm fine with it provided we have cause to use it, I just want to make sure we've thought about alternatives and not used it as default (cause it is so much easier to write tests that query test ids)

**Comment by zack** [comment]
@Midge-dev can i get another review please? :) thanks!

**Review by Midge-dev — APPROVED** [review]
LGTM :call_me_hand:

**Review by Dr-Pongo — APPROVED** [review]
FSR changes look solid! 🚀 TY!

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (Mottie) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:387]: [This icon was just removed last month in this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/25465)

Comment (Mottie) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:387]: Oh, hmmm... I see the screen shots. I thought the icon was part of the `va-alert`

**Review by CBonade — APPROVED** [review]

**Comment by zack** [comment]
Hello teams, could I please get a review?

@department-of-veterans-affairs/benefits-team-1-frontend
@department-of-veterans-affairs/platform-design-system-fe
@department-of-veterans-affairs/my-education-benefits
@department-of-veterans-affairs/1010-health-apps-frontend
@department-of-veterans-affairs/govcio-vfep-codereviewers
@department-of-veterans-affairs/platform-va-product-forms

Merging is blocked until all code owners have reviewed.

Thank you!

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (Mottie) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:328]: I think it's a typo LOL... I think "push" is a typo as well.

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (Mottie) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:393]: - "form-review-panel" and "usa-accordion-bordered' seem to be used by a few apps for styling
- There is CSS for "schemaform-review-chapter-error" (and the associated icon) can be removed

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (Mottie) [inline src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:73]: I think these links need to remain as links. If you look at the [reviewErrors docs](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/docs/reviewErrors.md) you'll see a screenshot of what this renders. Having these all as primary buttons in this list with a lot of text would be overwhelming for users.

Also, are we sure scrolling to a just-opened accordion works?

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (Mottie) [inline src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:76]: This work depends on being able to programmatically open an accordion. We should update it to click on the targeted accordion.

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:321]: `schemaform-chapter-accordion-content` is still used a class [here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/pre-need/components/CollapsiblePanel.jsx#L36), which is outside the scope of my work on this ticket so I don't want to touch it.

`usa-accordion-content` is also still used in a number of other places around the site, so those tests and styles will still be relevant for them:

```
src/applications/discharge-wizard/components/gpMinorComponents/AdditionalInstructions.jsx
src/applications/discharge-wizard/components/InstructionsPage.jsx
src/applications/proxy-rewrite/partials/header.js
src/applications/post-911-gib-status/components/EnrollmentPeriod.jsx
src/applications/edu-benefits/components/ConfirmationPageContent.jsx
src/applications/pre-need/components/CollapsiblePanel.jsx
src/applications/gi/components/AccordionItem.jsx
src/applications/gi/components/SearchAccordion.jsx
src/platform/site-wide/va-footer/components/MobileLinks.jsx
src/platform/site-wide/va-footer/components/LanguageSupport.jsx
src/platform/landing-pages/dev-template.ejs
```

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx:399]: This is still being used [here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/pre-need/components/CollapsiblePanel.jsx#L10).

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:76]: Good call. I'm going on code here as nothing I've tried is able to generate these links. The best I can do is a generic "something went wrong, call us" error message. Do you have instructions on how to get them to show up? It appears I need to give them more attention.

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (Mottie) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:73]: There is no utility class for styling buttons as links any more. Having these as links does break the material honesty convention, but I think having buttons with a lot of text inside is worse. We can bring this up with an a11y expert.

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (Mottie) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:76]: The [review errors docs are here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/docs/reviewErrors.md); they are currently implemented in the 526 (all-claims) form.

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:73]: @laflannery @jeana-adhoc 

Do either of you have a position on this? Both solutions here are easy, it's just a matter of which tradeoffs we want to take.

Comment (laflannery) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:73]: @zack and @Mottie I'm just trying to catch up - we are looking at [this error message](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/docs/images/error-text-based-on-error-name.png) correct? Where items such as "Has training pay" and "POW status" are interactive items that do something? I thought these were anchor links that take the user further down the page to the correct section

Comment (laflannery) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:73]: Thanks @Mottie. I think links would make sense here rather than buttons. I agree with your statement that having this many buttons probably is not going to be great. Also, thinking about this in terms of what is happening - when I activate these items - they are going to take me somewhere, navigate me to the particular section of the page/form where I need to go to edit and that navigational element is what a link is for rather than a button. So that's my reasoning behind why I think links would make sense here.

Comment (zack) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:73]: The more I try to generate these on the page the less convinced I am they even work? @Mottie maybe you can help me see what I'm missing here:

If we search for `ErrorLinks` in the codebase we get these results:

```
👾  vets-website(main) $ rg ErrorLinks
src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
8:const ErrorLinks = props => {
110:ErrorLinks.propTypes = {
121:)(ErrorLinks);
123:export { ErrorLinks };

src/platform/forms-system/src/js/review/submit-states/ValidationError.jsx
9:import ErrorLinks from './ErrorLinks';
23:    <ErrorLinks appType={appType} testId={testId} />

src/platform/forms-system/test/js/review/submit-states/ErrorLinks.unit.spec.jsx
7:import { ErrorLinks } from '../../../../src/js/review/submit-states/ErrorLinks';
9:describe('<ErrorLinks />', () => {
18:    const view = render(<ErrorLinks {...props} />);
42:    const view = render(<ErrorLinks {...props} />);
65:    const view = render(<ErrorLinks {...props} />);
87:    const view = render(<ErrorLinks {...props} />);
91:    view.rerender(<ErrorLinks {...props} errors={[]} />);
```

Which is one set of tests, one invocation, and one definition. Which is to say they're used in exactly one place.

In `ErrorLinks.jsx` the `errors` array that is used to generate the links in question comes from `props`, but in the _only_ place `ErrorLinks` is ever used (ValidationError.jsx#23), no errors are passed in, which means we'd see something like this:

image

I think the invocation on `ValidationError.jsx#23` needs to be passing `props.formErrors` into `ErrorLinks` using its `errors` prop. But with a regression like this I think it should also get tests. And maybe in that work we can come back to the styling of the buttons/links? It feels like some serious scope creep on this ticket which is supposed to be fixing a11y issues on the accordion and edit buttons.

Comment (zack) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:73]: Deleted my last message because I'm still trying to remember that we're using redux and `mapStateToProps`. 😨

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (Mottie) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:76]: The easiest way to get the error links to show up is to start a new form 526, then use the save-in-progress menu to jump to the review & submit page, then submit the form

Comment (zack) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:76]: Ah thanks, that's a lot easier than the solution I came up with :)))

File: src/platform/forms-system/src/js/utilities/review/index.js
Comment (Mottie) [inline src/platform/forms-system/src/js/utilities/review/index.js:53]: Can we pass in the pageKey to this function? This would only target the first page inside a chapter, which may not be the page that needs to be edited.

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (Mottie) [inline src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx:78]:
```suggestion
                          openAndEditChapter(error.chapterKey, error.pageKey);
```

File: src/platform/forms-system/src/js/utilities/review/index.js
Comment (zack) [reply src/platform/forms-system/src/js/utilities/review/index.js:53]: Great catch. I was hoping to get all the tests green before pinging you for a followup, but I see you're just on top of things. :)


https://github.com/department-of-veterans-affairs/vets-website/assets/2482403/53d70dfc-e977-4e3d-b397-30c9a50cf14b

File: src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/submit-states/ErrorLinks.jsx]: Links are back :)

**Comment by zack** [comment]
Good morning @Mottie, hope you had a nice weekend :) Is there anything else you'd like to discuss on the PR?

Thanks!

File: src/platform/forms-system/src/js/review/ObjectField.jsx
Comment (Mottie) [reply src/platform/forms-system/src/js/review/ObjectField.jsx:165]: This class name still needs to be added. As I mentioned, some [apps & tests use this class name](https://github.com/search?q=repo%3Adepartment-of-veterans-affairs%2Fvets-website%20.edit-btn&type=code)

**Comment by Mottie** [comment]
Thanks for following up! I added one more comment about the `edit-btn` class, but everything else looks good.

File: src/platform/forms-system/src/js/review/ObjectField.jsx
Comment (zack) [reply src/platform/forms-system/src/js/review/ObjectField.jsx:165]: Weird, I remember doing this...

Comment (zack) [reply src/platform/forms-system/src/js/review/ObjectField.jsx:165]: Oh, you know what, no, I tried it but there's no way to add class names to `va-button`.

I think we'll have to go look at each case of its use individually and see if it's referring to this page or another page (since the class 'edit-btn' is used elsewhere as well) and fix it if it's referring to this page.

Comment (zack) [reply src/platform/forms-system/src/js/review/ObjectField.jsx:165]: Ok so I don't think any of these are related:

```
👾  vets-website(dbex-64149-button-contrast)± $ rg '\.edit-btn'                                                                       
src/applications/financial-status-report/sass/financial-status-report.scss
356:@include hide-button(".edit-btn.primary-outline");

src/applications/appeals/testing/components/AreaOfDisagreement.jsx
107:        `[name="areaOfDisagreementFollowUp${pagePerItemIndex}ScrollElement"] + form .edit-btn`,

src/applications/appeals/shared/components/AreaOfDisagreement.jsx
106:        `[name="areaOfDisagreementFollowUp${pagePerItemIndex}ScrollElement"] + form .edit-btn`,

src/applications/edu-benefits/10203/tests/edu-10203.cypress.spec.js
79:        cy.get('.edit-btn').click({ multiple: true });

src/applications/hca/tests/components/FormReview/DependentsReviewPage.unit.spec.js
22:      const selector = container.querySelector('.edit-btn');
55:      const selector = container.querySelector('.edit-btn');
85:      const selector = container.querySelector('.edit-btn');

src/platform/forms-system/src/js/review/ArrayField.jsx
219:      this.scrollToAndFocus(`${fieldName}-${index}`, '.edit-btn');
```

`financial-status-report.scss`: is scoped to `.form-review-panel`, of which there is only one on the entire site, and it's not on this page. In fact, it's on another custom accordion that will need to be replaced some day. I do wonder if it's a typo, because there are dozen if not hundreds of `'.form-review-panel-page'`, but that's so far out of scope it's not something I'm going to worry about!

`AreaOfDisagreement.jsx` (both of them): Yep, fixed these!

`edu-10203.cypress.spec.js`: This is fine since it's a passing cypress test. If this was referring to the `<va-button>` as opposed to some other button with the same class then the `.click` would throw an exception.

`DependentsReviewPage.unit.spec.js`: Also a passing test. All of these queries are explicitly tested as existing.

`ArrayField.jsx`: This refers to some code that refocuses the edit button after editing an array element. This is a feature of the review page and is working, so this is not being used here, and is therefor referring to some other page.

This `ObjectField.jsx` is _only_ imported by `SchemaForm.jsx` (plus some tests) for review fields, so I don't believe we'll see it popping up on pages other than this review page, which means the scope is quite limited.

I know this is a bit all over the place, let me know what you think! I'm pretty confident in it.

Comment (zack) [reply src/platform/forms-system/src/js/review/ObjectField.jsx:165]: Good morning @Mottie, how are you feeling about this?

**Review by ataker — APPROVED** [review]

**Review by GovNapoleon — APPROVED** [review]

**Review by longmd — APPROVED** [review]

**Review by jadon1979 — APPROVED** [review]

**Review by Mottie — APPROVED** [review]

**Review by ndsprinkle — APPROVED** [review]

**Review by bradbergeron-us — APPROVED** [review]

**Pull request merged**
